bindings/binding_core_wasm/__tests__/error.js

```diff
@@ -1,9 +1,6 @@
 const swc = require("../pkg");
 
-it("properly reports error", function () {
-    expect(() => {
-        swc.transformSync("Foo {}", {});
-    }).toThrow("Syntax Error");
+it("properly reports error", () => {
```

Comment:
Please revert this. The `SyntaxError` type includes the text `Syntax Error`

Reply:
Got it, i should add `Syntax Error` text in js side `@swc/core` package.

Reply:
To be exact, we should throw the JS SyntaxError type insted of adding text

Reply:
I found it's hard to do it, js side just receive a error from napi. But js do not known what kind error is.

Reply:
`napi` has an API to throw syntax errors.
https://docs.rs/napi/latest/napi/struct.Env.html#method.throw_syntax_error

Reply:
I have a idea that we maintain syntax error in `try_with_handler`, we only need find a ways to control print formatting way.
For example: continue using `Debug` trait and `Display` trait to control the print formatting. 
I found swc using `format!("{}", err)` or `format!("{:?}", err)` to control formatting before this PR.

Reply:
Do you mean we should include Syntax Error text from diagnostics from the swc crate?

Reply:
Yes.Stay the same as before. If we need report syntax Error in js not rust, maybe we need have other PR.
And another reason is Syntax Error emit by js side the mainly aim is to resolve the print formatting.

Reply:
SyntaxError should be thrown from Rust code using napi

Reply:
Got it, i will update it later

Reply:
I remove extra error struct. 
And I kept the original writing, continue using anyhow::Error::context o ensure print correctly, reference to https://github.com/swc-project/swc/blob/main/crates/swc_error_reporters/src/handler.rs#L153